src/util/duration.h

```diff
@@ -62,6 +63,11 @@ class DurationBase {
         return static_cast<double>(m_durationNanos);
     }
 
+    // Returns the duration as an integer number of nanoseconds.
+    constexpr double toFrames(mixxx::audio::SampleRate sampleRate) const {
```

Comment:
The *duration* should not know anything about *frames* and *sample rates*.

Reply:
Writing `duration.toDoubleSeconds() * sampleRate` is tedious and not that expressive, so my initial idea was to "express a duration in frames". But I agree that a frame count is not a during without knowing the sample rate.

The alternative would be to add a method to the `SampleRate` class instead:

```cpp
constexpr double SampleRate::frameCount(mixxx::Duration duration) {
    return duration.toDoubleSeconds() * value();
}
```

And using it would look like this:

```cpp
constexpr auto bpm = mixxx::Bpm(120);
constexpr auto sampleRate = mixxx::audio::SampleRate(44100);

const double beatLengthFrames = sampleRate.frameCount(bpm.beatLength());
```

What do you think?

Reply:
Conceptual Dependency Graph: Duration <- Sample Rate <- Frame

Sample rate knows about duration because of frequencies. But it doesn't need to know anything about *frames*.

Frames know about sampling because their purpose is to represent a continuous audio signal as discrete, sampled values.

Therefore any calculation that involves frames should not taint neither duration nor sample rate code.

Reply:
Why not `double operator*(Duration, SampleRate)` (and its twin with switched arguments) in `samplerate.h`? The unit of the resulting value depends on the context.